### ISSUE TITLE
Operating Photoprism behind authentication proxy with PHOTOPRISM_PUBLIC=true

### ISSUE THREAD

**Issue by voigt** [photoprism/photoprism#811]
Hi Photoprism Team,

first of all I'd like to thank you for all of your efforts resulting into such a great product!

I'm currently evaluating Photoprism for personal use and intend to run it behind [Pomerium](https://github.com/pomerium/pomerium) as an authentication proxy (simply as I secure most of my personal services with it). The entire setup roughly depends on: Traefik -> Pomerium -> Photoprism.

This works just fine - I set `PHOTOPRISM_PUBLIC=true`, so Photoprism doesn't need to handle any auth.

Now I discovered, that Photoprism provides a sharing functionality which exposes Albums via `<host>/s/<id>`. Therefore I excluded the path `/s` from authentication - the Pomerium [policy](https://www.pomerium.io/reference/#policy) looks something like this:

```
policy:
  - from: https://traefik.christophvoigt.com
    to: http://0.0.0.0:8080
    allowed_users:
      - <...>
  - from: https://photo.christophvoigt.com
    to: http://0.0.0.0:2342
    prefix: /s
    allow_websockets: true
    allow_public_unauthenticated_access: true
  - from: https://photo.christophvoigt.com
    to: http://0.0.0.0:2342
    allow_websockets: true
    allowed_users:
      - <...>
```

The rule works as expected, unfortunately the shared album does not load properly:

<img width="964" alt="grafik" src="https://user-images.githubusercontent.com/2982698/103529412-e97f8f00-4e85-11eb-8187-2dd9da4923e9.png">

As the reason I identified a request to `/api/v1/session`.

<img width="1392" alt="grafik" src="https://user-images.githubusercontent.com/2982698/103529746-82160f00-4e86-11eb-9f0c-7bdbb74520d2.png">

Allowing `/api/v1/session` will lead to an additional request to `/api/v1/albums`. Allowing this path obviously exposes more than I want :)

I'm totally aware, that Photoprism actually relies on its own auth for sharing (so this is not even an edge case). Though I'd kindly ask for answers to the following questions:

1. Is there any option I might have overseen? I thought about running an additional read-only instance of Photoprism, working with the same database - is there anything obvious why this would not work?
3. Are there any plans to deliver prerendered albums that do not rely on additional authenticated api-requests from the client?

Otherwise, I simply need to wait for #782 and thank you for your time :)

**Comment by lastzero** [comment]
Our UI is a progressive Web app. What you exposed via `/s/...` is just bootstrapping HTML, no data, no logic. There is one common API - no external software can manage permissions down to specific actions, methods, users, and entities.

I guess what you want / need / expect is a static sharing page including all images and metadata, so that no API and no API auth is needed. Not sure how video streaming should be implemented in this case?

Bottom line: We don't offer this feature at the moment, and don't have the resources to develop it right now. Especially since there was no demand. When we reach more than 3% of our funding goal and there is demand, we may develop this for you and others who prefer to take care of application security on their own.

**Comment by lastzero** [comment]
Is static sharing something you would like to see developed and are willing to sponsor, or shall we close this issue?

**Comment by benmccann** [comment]
I took a look at Pomerium. It seems it was originally forked from Bitly's oauth2_proxy (which is no longer being maintained) and there are other forks as well such as [buzzfeed/sso](https://github.com/buzzfeed/sso) and [oauth2-proxy/oauth2-proxy](https://github.com/oauth2-proxy/oauth2-proxy). None of them have super clear docs, but from what I can see there are probably better ways to implement this than using a static HTML page.

`buzzfeed/sso` and `oauth2-proxy` add an `X-Forwarded-User` and `X-Forwarded-Email` to the request and then sign the request. [Pomerium adds an `X-Pomerium-Jwt-Assertion` header with signed attestation JWT](https://www.pomerium.io/docs/topics/getting-users-identity.html#verification). We should be able to use these to automatically log the user in. That would give you full access to all the functionality including sharing photos with other users after multi-user support exists.
